polaris-core/src/main/java/org/apache/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java

```diff
@@ -58,11 +60,17 @@ public abstract class LocalPolarisMetaStoreManagerFactory<StoreType>
   private static final Logger LOGGER =
       LoggerFactory.getLogger(LocalPolarisMetaStoreManagerFactory.class);
 
+  private final PrincipalSecretsGenerator.Realms secretsGenerator = bootstrap();
```

Comment:
This looks a bit odd, why is a `secretsGenerator` of the type `Realms`?

Reply:
I see, it's like a `secretsGeneratorGenerator`. I wonder if we can just get rid of the thin `Realms` type and push this into `PrincipalSecretsGenerator`

Reply:
I believe this will require `PolarisMetaStoreSession` implementations to have "realm" as a field. Would that be ok from your POV?.. Currently sessions are not strongly bound to realms. Realm-specific behaviour is relevant only to the `bootstrap` command and factories.

Reply:
I was wondering if we could just push `realm` into `produceSecrets` as an argument

Reply:
I see, the places calling `produceSecrets` don't necessarily have access to the `realm`. Maybe we can just move this into the constructor of `PrincipalSecretsGenerator` itself

Reply:
I reworked the impl. a bit... hopefully the bootstrapping logic is clearer now.

@eric-maynard: Would you still prefer to have a non-lambda class for `PrincipalSecretsGenerator` implementations?

Reply:
I think it's fine as it is now, maybe in the future if there are more implementations we can refine it further. Thanks for iterating on this!